docs/source/error_code_list2.rst

```diff
@@ -0,0 +1,166 @@
+type of a cast is the same as the target type.
+
+Example:
+
+.. code-block:: python
+
+    # mypy: warn-redundant-casts
+
+    from typing import cast
+
+    Count = int
+
+    def example(x: Count) -> int:
+        # Error: Redundant cast to "int"  [redundant-cast]
+        return cast(int, x)
+
+Check that comparisons are overlapping [comparison-overlap]
+-----------------------------------------------------------
+
+If you use ``--strict-equality``, mypy will generate an error if it
+thinks that a comparison operation is always true or false. These are
+often bugs. Sometimes mypy is too picky and the comparison can
+actually be useful. Instead of disabling strict equality checking
+everywhere, you can use ``# type: ignore[comparison-overlap]`` to
+ignore the issue on a particular line only.
+
+Example:
+
+.. code-block:: python
+
+    # mypy: strict-equality
+
+    def is_magic(x: int) -> bool:
+        # Error: Non-overlapping equality check (left operand type: "int",
+        #        right operand type: "str")  [comparison-overlap]
+        return x == 'magic'
+
+Check that no untyped functions are called [no-untyped-call]
+------------------------------------------------------------
+
+If you use ``--disallow-untyped-calls``, mypy generates an error when you
+call an unannotated function in an annotated function.
+
+Example:
+
+.. code-block:: python
+
+    # mypy: disallow-untyped-calls
+
+    def do_it() -> None:
+        # Error: Call to untyped function "bad" in typed context  [no-untyped-call]
+        bad()
+
+    def bad():
+        ...
+
+
+Check that function does not return Any value [no-any-return]
+-------------------------------------------------------------
+
+If you use ``--warn-return-any``, mypy generates an error if you return a
+value with an ``Any`` type in a function that is annotated to return a
+non-``Any`` value.
+
+Example:
+
+.. code-block:: python
+
+    # mypy: warn-return-any
+
+    def fields(s):
+         return s.split(',')
+
+    def first_field(x: str) -> str:
+        # Error: Returning Any from function declared to return "str"  [no-any-return]
+        return fields(x)[0]
+
+Check that types have no Any components due to missing imports [no-any-unimported]
+----------------------------------------------------------------------------------
+
+If you use ``--disallow-any-unimported``, mypy generates an error if a component of
+a type becomes ``Any`` because mypy couldn't resolve an import. These "stealth"
+``Any`` types can be surprising and accidentally cause imprecise type checking.
+
+In this example, we assume that mypy can't find the module ``animals``, which means
+that ``Cat`` falls back to ``Any`` in a type annotation:
+
+.. code-block:: python
+
+    # mypy: disallow-any-unimported
+
+    from animals import Cat  # type: ignore
+
+    # Error: Argument 1 to "feed" becomes "Any" due to an unfollowed import  [no-any-unimported]
+    def feed(cat: Cat) -> None:
+        ...
```

Comment:
What error codes are used for `--disallow-subclassing-any` or `--disallow-any-decorated` or the several other similar flags?

Reply:
They use `misc` at the moment. Created an issue about this: #7457